### PR TITLE
Use Jekyll for posts, indexes and sitemap

### LINKED ISSUE #70: [META] L'indexeur est nul
L'indexeur est naze, on se retrouve souvent avec des diffs qui n'ont rien à voir avec les articles en cours de modification et on passe notre temps à essayer de réparer les problèmes.

On a eu diverse discussions sur discord, on a déjà proposé d'arrêter de sortir l'information depuis l'historique git, de sauvegarder (dupliquer donc) l'information à part dans un fichier tout propre qu'on pourrait gérer nous-même.

Ensuite @Equinox4 a remarqué qu'il y avait déjà une fonctionnalité d'indexeur dans jekyll et qu'il suffirait peut-être de juste l'utiliser plutôt que de faire de la merde avec nos outils fait maison

Enfin bref, voila une issue pour centraliser nos discussion

### PR DESCRIPTION
**Pull request by AllanBlanchard**
Close https://github.com/NotANameServer/Not-a-Hub/issues/70

### CODE REVIEW

File: _includes/date.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{% comment %}
+  We use this file to print out dates because GitHub pages does not support the
+  Jekyll plugin dedicated to locales ...
+{% endcomment %}
+
+{% assign m = include.the_date | date: "%-m" %}
+{{ page.date | date: "%-d" }}
```
Comment (Elanis) [inline _includes/date.html]:
```suggestion
{{ include.the_date | date: "%-d" }}
```

File: _includes/date.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  {% when '5' %}Mai
+  {% when '6' %}Juin
+  {% when '7' %}Juillet
+  {% when '8' %}Août
+  {% when '9' %}Septembre
+  {% when '10' %}Octobre
+  {% when '11' %}Novembre
+  {% when '12' %}Decembre
+{% endcase %}
+{{ page.date | date: "%Y" }}
```
Comment (Elanis) [inline _includes/date.html]:
```suggestion
{{ include.the_date | date: "%Y" }}
```

**Review by Elanis — CHANGES_REQUESTED** [review]

File: game-dev/ressources.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
-# Ressources pour game-dev
+---
+layout: post
+author: Le serveur Not a Name
+date: 2021-09-04
+last_update: 2023-04-12
```
Comment (Elanis) [inline game-dev/ressources.md]:
```suggestion
last_update: 2023-03-12
```

Comment (Elanis) [reply game-dev/ressources.md]: https://github.com/NotANameServer/Not-a-Hub/commits/master/game-dev/ressources.md

File: langages/scala/programmer-en-scala-3-booleens-et-conditions.md [hunk 4]
```diff
@@ -1,4 +1,10 @@
-# Programmer en Scala 3 : Booléens et conditions
+---
+layout: post
+author: Raphaël Fromentin et TheElectronWill
+date: 2022-09-16
```
Comment (Elanis) [inline langages/scala/programmer-en-scala-3-booleens-et-conditions.md]:
```suggestion
date: 2023-03-10
```

Comment (Elanis) [reply langages/scala/programmer-en-scala-3-booleens-et-conditions.md]: https://github.com/NotANameServer/Not-a-Hub/commits/master/langages/scala/programmer-en-scala-3-booleens-et-conditions.md

**Review by Elanis — COMMENTED** [review]
Je pense que ces deux dates sont fausses

**Review by Elanis — APPROVED** [review]

**Review by Julien00859 — APPROVED** [review]

**Review by Elanis — APPROVED** [review]

**Pull request merged**